controllers/database.go

```diff
@@ -31,6 +36,53 @@ var dbTemplates = []string{
 	dbSecret,
 }
 
+// extract to var for mocking in testing
+var ConnectAndQueryDatabase = func(host, port, username, password, dbname string) bool {
+	connectionString := fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", username, password, host, port, dbname)
+	db, err := sql.Open("mysql", connectionString)
+	if err != nil {
+		return false
+	}
+	defer db.Close()
+
+	testStatement := "SELECT 1;"
+	_, err = db.Exec(testStatement)
+	return err == nil
+}
+
+func (r *DSPAReconciler) VerifyMySQLDBConnection(host, port, username, password, dbname string) bool {
+	return ConnectAndQueryDatabase(host, port, username, password, dbname)
+}
+
+func (r *DSPAReconciler) isDatabaseAccessible(ctx context.Context, dsp *dspav1alpha1.DataSciencePipelinesApplication,
+	params *DSPAParams) bool {
+	log := r.Log.WithValues("namespace", dsp.Namespace).WithValues("dspa_name", dsp.Name)
+
+	log.Info("Performing Database Health Check")
+	databaseSpecified := dsp.Spec.Database != nil
+	usingExternalDB := params.UsingExternalDB(dsp)
+	usingMariaDB := !databaseSpecified || dsp.Spec.Database.MariaDB != nil
+	if usingMariaDB || usingExternalDB {
+		decodePass, _ := b64.StdEncoding.DecodeString(params.DBConnection.Password)
+		db_connect := r.VerifyMySQLDBConnection(params.DBConnection.Host,
+			params.DBConnection.Port,
+			params.DBConnection.Username,
+			string(decodePass),
+			params.DBConnection.DBName)
+		if db_connect {
+			log.Info("Database Health Check Successful")
+		} else {
+			log.Info("Unable to connect to Database")
+		}
+		return db_connect
+
+	}
+
+	log.Info("Could not connect to Database: Unsupported Type")
+	// Only MariaDB and Mysql-Compliant Database supported.
```

Comment:
this comment looks like something that doesn't belong here, and is better added [here](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/api/v1alpha1/dspipeline_types.go#L122). 

If you really want to add it here, then I think we should add it as a function  docstring/comment right above it.

Reply:
moved the conditionals around, so this code and comment now should make a litle more sense